src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -416,11 +409,42 @@ private JButton getPrintButton() {
 		jButtonPrint.setEnabled(false);
 		return jButtonPrint;
 	}
-	
+
 	private JButton getDeleteButton() {
 		jButtonDelete = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
 		jButtonDelete.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
 		jButtonDelete.setEnabled(false);
+
+		jButtonDelete.addActionListener(actionEvent -> {
+			int selectedRow = jTableInventory.getSelectedRow();
+			if (selectedRow == -1) {
+				MessageDialog.error(this, MessageBundle.getMessage("angal.inventory.pleaseselectinventory.msg"));
+				return;
+			}
+
+			MedicalInventory inventory = inventoryList.get(selectedRow);
+			String status = inventory.getStatus().toLowerCase();
+
+			if ("draft".equals(status) || "validated".equals(status)) {
+				int confirm = JOptionPane.showConfirmDialog(this,
```

Comment:
Why didn't you use `org.isf.utils.jobjects.MessageDialog.java` and the static method `yesNo`?  See https://github.com/informatici/openhospital-gui/blob/d758e134bf9070c4f4c9f72a08dfad22c2f26b85/src/main/java/org/isf/utils/jobjects/MessageDialog.java#L86

Reply:
Thanks for the suggestions.